edx-solutions/edx-platform
# MCKIN-8918 Implement call to discussion's user retirment

This PR implements calling the user retirement feature in discussions.

Also, api-integration is bumped to 2.5.2 to make this available to custom frontend.

**Testing instructions**:

1. Create a test user with access to a course that contains discussions.
2. Login with the test user and create posts and comments.
3. Back to the admin user, use the custom frontend UI to delete the test user.
4. With another user with access to the course, make sure the created posts and comments were properly anonymized, having title, contents and author overwritten.

![image](https://user-images.githubusercontent.com/5691347/52211323-b2a33780-2870-11e9-92c6-a51b1576fe17.png)

BONUS:

You can check no information about the user is left in the MongoDB collections used by discussions:
```bash
vagrant@vagrant:~$ mongo
MongoDB shell version: 2.6.12
connecting to: test
> use cs_comments_service
switched to db cs_comments_service
> db.contents.find({author_username: "test_user"})
> db.users.find({username: "test_user"})
> db.users.find({username: "Deleted user"})
{ "_id" : "41", "default_sort_key" : "activity", "external_id" : "41", "username" : "Deleted user", "read_states": [ ], "email" : "" }
```

**Reviewers**
